test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,266 @@
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openColorsPanel();
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] .component-color-indicator'
+			)
+		).toHaveCSS( 'background', /rgb\(255, 239, 11\)/ );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors text styles"] .component-color-indicator'
+			)
+		).toHaveCSS( 'background', /rgb\(25, 25, 17\)/ );
+
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openTypographyPanel();
+		await siteEditorStyleVariations.openTextPanel();
+
+		expect( await siteEditorStyleVariations.getFontSizeHint() ).toBe(
+			'(Custom)'
+		);
+		expect( await siteEditorStyleVariations.getCustomFontSizeValue() ).toBe(
+			'15'
+		);
+	} );
+
+	test( 'should apply a color palette in a variation', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openColorsPanel();
+		await page.click( `role=button[name="Color palettes"i]` );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Foreground"i]' )
+		).toHaveCSS( 'background', /rgb\(74, 7, 74\)/ );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Background"i]' )
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Awesome pink"i]' )
+		).toHaveCSS( 'background', /rgba\(204, 0, 255, 0\.77\)/ );
+	} );
+
+	test( 'should reflect style variations in the styles applied to the editor', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		const frame = page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
+		const paragraph = frame.locator(
+			'xpath=//p[text()="My awesome paragraph"]'
+		);
+		const paragraphColor = await paragraph.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).color;
+		} );
+		expect( paragraphColor ).toBe( 'rgb(25, 25, 17)' );
+		const body = frame.locator( 'css=body' );
+		const backgroundColor = await body.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).backgroundColor;
+		} );
+		expect( backgroundColor ).toBe( 'rgb(255, 239, 11)' );
```

Comment:
Same here. Let's use `toHaveCSS`.

```suggestion
		await expect( body ).toHaveCSS( 'background-color', 'rgb(255, 239, 11)' );
```